### ISSUE TITLE
solver_opts are not copied by sklearn clone

### ISSUE THREAD

**Issue by lbluque** [CederGroupHub/sparse-lm#12]
tldr; pass `solver_opts` as a dictionary and not variable kwargs.

cvxpy solver options are passed as variable keyword arguments `kwargs` and saved as an attribute `self.solver_opts` 

This works without issue when running `estimator.solve` directly.

But when using a model selection tool from `sklearn` such as `GridSearchCV`, any options saved in `self.solver_kwargs` are not used. This happens due to how `BaseSearchCV` makes a clone of the estimator: it copies using `get_params`, and so it does not copy any variable keyword arguments:

https://github.com/scikit-learn/scikit-learn/blob/36958fb240fbe435673a9e3c52e769f01f36bec0/sklearn/base.py#L177-L180

The suggested fix is to simply pass `solver_opts` as a dictionary.
